src/test/java/com/neo4j/docker/neo4jserver/TestHealthcheck.java

```diff
@@ -0,0 +1,112 @@
+import com.neo4j.docker.neo4jserver.configurations.Setting;
+import com.neo4j.docker.utils.TemporaryFolderManager;
+import com.neo4j.docker.utils.TestSettings;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+import static java.lang.String.format;
+
+public class TestHealthcheck
+{
+    private static Logger log = LoggerFactory.getLogger( TestHealthcheck.class );
+    private static Map<Setting,Configuration> confNames = Configuration.getConfigurationNameMap();
+
+    @RegisterExtension
+    public static TemporaryFolderManager temporaryFolderManager = new TemporaryFolderManager();
+
+    private GenericContainer createContainerWithDefaultListenAddress()
+    {
+        GenericContainer container = new GenericContainer( TestSettings.IMAGE_ID );
+        container.withEnv( "NEO4J_ACCEPT_LICENSE_AGREEMENT", "yes" )
+                 .withExposedPorts( 7474, 7687 )
+                 .withLogConsumer( new Slf4jLogConsumer( log ) );
+        return container;
+    }
+
+    private GenericContainer createContainerWithEnvVarListenAddress( String listenAddress )
+    {
+        GenericContainer container = new GenericContainer( TestSettings.IMAGE_ID );
+        container.withEnv( "NEO4J_ACCEPT_LICENSE_AGREEMENT", "yes" )
+                 .withEnv( confNames.get( Setting.HTTP_LISTEN_ADDRESS ).envName, listenAddress )
+                 .withExposedPorts( 7474, 7687 )
+                 .withLogConsumer( new Slf4jLogConsumer( log ) );
+        return container;
+    }
+
+    private GenericContainer createContainerWithConfigListenAddress( String listenAddress ) throws IOException
+    {
+        GenericContainer container = new GenericContainer( TestSettings.IMAGE_ID );
+        container.withEnv( "NEO4J_ACCEPT_LICENSE_AGREEMENT", "yes" )
+                 .withExposedPorts( 7474, 7687 )
+                 .withLogConsumer( new Slf4jLogConsumer( log ) );
+
+        var tempConfigDir = temporaryFolderManager.createTempFolder( "temp_neo4j_config" );
+
+        var neo4jConfig = new File( tempConfigDir.toAbsolutePath() + File.separator + "neo4j.conf" );
+        var writer = new BufferedWriter( new FileWriter( neo4jConfig ) );
+        Map<Setting,Configuration> confNames = Configuration.getConfigurationNameMap();
+        writer.write( format( "%s=%s", confNames.get( Setting.HTTP_LISTEN_ADDRESS ).name, listenAddress ) );
+        writer.close();
```

Comment:
you can replace 62-66 with just:
```java
Files.write( tempConfigDir.resolve( "neo4j.conf" ), 
    format( "%s=%s", confNames.get( Setting.HTTP_LISTEN_ADDRESS ).name, listenAddress ).getBytes());
```

also not sure why you're creating `confNames` again when it's already a member variable in the class.

Reply:
You can take the boy out of C++, but can't take C++ out of the boy

Reply:
The confNames init is left over from Ctrl+c Ctrl+v